src/gnatss/harmonic_mean.py

```diff
@@ -67,9 +145,35 @@ def sv_harmonic_mean(svdf: pd.DataFrame, start_depth: float, end_depth: float) -
         raise ValueError("Dataframe is empty! Please check your data inputs.")
     # Clean up the sound speed value, ensuring that there's no negative value
     svdf = svdf[svdf[SP_SOUND_SPEED] > 0].reset_index(drop=True)
-    # Make all of the values absolute values, so we're only dealing with positives
-    abs_start = abs(start_depth)
-    abs_end = abs(end_depth)
-    abs_sv = abs(svdf)
 
-    return _compute_hm(abs_sv, abs_start, abs_end)
+    for col in [SP_DEPTH, SP_SOUND_SPEED]:
+        if col not in svdf.columns:
+            raise ValueError(f"{col} column must exist in the input dataframe!")
+
+    if method == "scipy":
+        # Make all of the values absolute values, so we're only dealing with positives
+        abs_start = abs(start_depth)
+        abs_end = abs(end_depth)
+        abs_sv = abs(svdf)
+
+        svhm = _compute_hm(abs_sv, abs_start, abs_end)
+    elif method == "numba":
+        # Ensure both start and end depths are negative
+        start_depth = -np.abs(start_depth)
+        end_depth = -np.abs(end_depth)
+
+        if start_depth < end_depth:
+            raise ValueError("Start depth must be greater than end depth!")
+
+        svdf = svdf[(svdf[SP_DEPTH].round() <= start_depth)]
+        # Extract the numpy arrays for depth and sound speed
+        # Make sure that the depth array has negative values
+        # and the sound speed array has positive values
+        dd = -np.abs(svdf[SP_DEPTH].values)
+        sv = np.abs(svdf[SP_SOUND_SPEED].values)
+
+        svhm = _sv_harmon_mean(dd, sv, start_depth, end_depth)
+    else:
```

Comment:
Question for my own understanding.
Who enforces that only the valid literal values are used? Is that mypy or a runtime check. Ie. at runtime could we actually get here?

Reply:
If the `method` value is neither a "scipy" or "numba", some other string would go to the else statement. Currently a runtime check happens but there's typehints and documentation of the valid values.